### PR TITLE
faraz/LP-1537/LP-1538_provision_philu_code_in_ironwood

### PR DESCRIPTION
**Pull request by ahmed-faraz46**
- resolve circular imports
- remove url patterns becuase django 1.11 removed patterns
- move survey_gizmo app from philu to ironwood
- Remove usage of following methods as ironwood removed them
	- set_enterprice_branding_filters
        - dog_stats_wrapper
- remove a migration commerce because it adds fields in table which was removed by ironwood

### CODE REVIEW

File: common/djangoapps/student/models.py [hunk 1]
```diff
@@ -655,6 +654,9 @@ def user_post_save_callback(sender, **kwargs):
 
     Additionally, emit analytics events after saving the User.
     """
+
+    from lms.djangoapps.philu_overrides.tasks import task_enroll_user_in_pending_courses
```
Comment (emraan004) [inline common/djangoapps/student/models.py]: Let's document this core change

File: lms/asgi.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
-
-os.environ['DJANGO_SETTINGS_MODULE'] = "lms.envs.aws"
-os.environ['SERVICE_VARIANT'] = "lms"
-channel_layer = channels.asgi.get_channel_layer()
+# import os
+# import channels
+#
+# os.environ['DJANGO_SETTINGS_MODULE'] = "lms.envs.aws"
+# os.environ['SERVICE_VARIANT'] = "lms"
+# channel_layer = channels.asgi.get_channel_layer()
```
Comment (emraan004) [inline lms/asgi.py]: Let's uncomment them

File: lms/djangoapps/courseware/views/views.py [hunk 3]
```diff
@@ -341,6 +341,17 @@ def get_last_accessed_courseware(course, request, user):
                 return url
         return None
 
+
```
Comment (emraan004) [inline lms/djangoapps/courseware/views/views.py]: Let's recheck this piece of code

File: lms/djangoapps/philu_overrides/migrations/0001_initial.py [hunk 4]
```diff
@@ -11,14 +11,10 @@ class Migration(migrations.Migration):
...
         cursor = connection.cursor()
-        tables = ['enterprise_historicalenterprisecustomer', 'enterprise_historicaluserdatasharingconsentaudit']
-        for t in tables:
-            query = 'ALTER TABLE %s ADD start_date DATETIME NULL' % t
-            cursor.execute(query)
-            query = 'ALTER TABLE %s ADD end_date DATETIME NULL' % t
-            cursor.execute(query)
-            query = 'ALTER TABLE %s ADD history_change_reason VARCHAR(100) NULL' % t
-            cursor.execute(query)
+        query = 'ALTER TABLE enterprise_historicalenterprisecustomer ADD start_date DATETIME NULL'
```
Comment (emraan004) [inline lms/djangoapps/philu_overrides/migrations/0001_initial.py]: Check if we can execute single query for both columns

File: lms/djangoapps/philu_overrides/views.py [hunk 5]
```diff
@@ -83,7 +79,12 @@ def login_and_registration_form(request, initial_mode="login", org_name=None, ad
 
     """
     # Determine the URL to redirect to following login/registration/third_party_auth
-    _local_server_get('/user_api/v1/account/registration/', request.session)
+    from openedx.features.student_account.views import local_server_get, _get_form_descriptions, \
```
Comment (emraan004) [inline lms/djangoapps/philu_overrides/views.py]: verify this change

File: openedx/features/student_account/views.py [hunk 6]
```diff
@@ -9,23 +9,56 @@
...
 from lms.djangoapps.onboarding.helpers import reorder_registration_form_fields, get_alquity_community_url
-from lms.djangoapps.student_account.views import _local_server_get, _external_auth_intercept
+
 from openedx.core.djangoapps.theming.helpers import is_request_in_themed_site
 
 AUDIT_LOG = logging.getLogger("audit")
 log = logging.getLogger(__name__)
 User = get_user_model()  # pylint:disable=invalid-name
 
+def local_server_get(url, session):
```
Comment (emraan004) [inline openedx/features/student_account/views.py]: Let's move this method to Philu Overrides

File: common/djangoapps/student/models.py (hunk 1 above)
Comment (ahmed-faraz46) [reply common/djangoapps/student/models.py]: Done in `Ironwood core change`

File: lms/djangoapps/courseware/views/views.py (hunk 3 above)
Comment (ahmed-faraz46) [reply lms/djangoapps/courseware/views/views.py]: done, it was a mistake in copy and pasting of the code.

File: lms/djangoapps/philu_overrides/migrations/0001_initial.py (hunk 4 above)
Comment (ahmed-faraz46) [reply lms/djangoapps/philu_overrides/migrations/0001_initial.py]: Yes we can, here is the modified code
`ALTER TABLE enterprise_historicalenterprisecustomer ADD start_date DATETIME NULL, end_date DATETIME NULL;`

**Comment by ahmed-faraz46** [comment]
@emraan004 requested changes are tackled in [this branch](https://github.com/philanthropy-u/edx-platform/tree/faraz/LP-1387_integrate_onboarding), so I'm merging this PR, we can verify changes in the mentioned branch.

**Pull request merged**